docs/rules/no-parameter-properties.md

```diff
@@ -0,0 +1,342 @@
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for this rule with no options at all:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+```
+
+### readonly
+Examples of **incorrect** code for the `{ "allows": ["readonly"] }` options:
+```ts
+class Foo {
+    constructor(private name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for the `{ "allows": ["readonly"] }` options:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+
+class Foo {
+    constructor(readonly name: string) {}
+}
+```
+
+### private
+Examples of **incorrect** code for the `{ "allows": ["private"] }` options:
+```ts
+class Foo {
+    constructor(readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for the `{ "allows": ["private"] }` options:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+
+class Foo {
+    constructor(private name: string) {}
+}
+```
+
+### protected
+Examples of **incorrect** code for the `{ "allows": ["protected"] }` options:
+```ts
+class Foo {
+    constructor(readonly name: string) {}
+}
+
+class Foo {
+    constructor(private name: string) {}
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for the `{ "allows": ["protected"] }` options:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+```
+
+### public
+Examples of **incorrect** code for the `{ "allows": ["public"] }` options:
+```ts
+class Foo {
+    constructor(readonly name: string) {}
+}
+
+class Foo {
+    constructor(private name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
```

Comment:
I added **accesibility + readonly** modifiers as separate options to avoid scenarios like 'I want to allow `protected`, `public` and `public readonly`, but no `protected readonly`'.

Does that make sense? it did when I was writing it, but  I'm open to suggestions...